Could not pass event SignChangeEvent to dynmap v3.0-SNAPSHOT-243

When I place a sign with enablesigns=true
Dynmap-3.0-beta-7-spigot.jar
spigot-1.14.4.jar
```
[16:27:26] [Server thread/ERROR]: Could not pass event SignChangeEvent to dynmap v3.0-beta-7-241
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:320) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:529) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:514) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:2381) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PacketPlayInUpdateSign.a(SourceFile:44) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PacketPlayInUpdateSign.a(SourceFile:10) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.aX(MinecraftServer.java:909) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:902) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:886) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:819) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
Caused by: java.lang.IllegalArgumentException: Cannot get ID of Modern Material
        at org.apache.commons.lang.Validate.isTrue(Validate.java:136) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.Material.getId(Material.java:3302) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.dynmap.bukkit.DynmapPlugin$BukkitServer$5.onSignChange(DynmapPlugin.java:369) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:316) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
```

Latest SNAPSHOT also has same problem.
```
[16:45:13] [Server thread/ERROR]: Could not pass event SignChangeEvent to dynmap v3.0-SNAPSHOT-243
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:320) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:529) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:514) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:2381) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PacketPlayInUpdateSign.a(SourceFile:44) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PacketPlayInUpdateSign.a(SourceFile:10) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.aX(MinecraftServer.java:909) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:902) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:886) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:819) [spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
Caused by: java.lang.IllegalArgumentException: Cannot get ID of Modern Material
        at org.apache.commons.lang.Validate.isTrue(Validate.java:136) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.bukkit.Material.getId(Material.java:3302) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        at org.dynmap.bukkit.DynmapPlugin$BukkitServer$5.onSignChange(DynmapPlugin.java:369) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:316) ~[spigot-1.14.4.jar:git-Spigot-56f8471-6567017]
        ... 17 more
```

--- a/DynmapCore/src/main/java/org/dynmap/DynmapCore.java
+++ b/DynmapCore/src/main/java/org/dynmap/DynmapCore.java
@@ -84,7 +84,7 @@ public class DynmapCore implements DynmapCommonAPI {
      */
     public static abstract class EnableCoreCallbacks {
         /**
-         * Called during enableCore to report that confniguration.txt is loaded
+         * Called during enableCore to report that configuration.txt is loaded
          */
         public abstract void configurationLoaded();
     }
@@ -2400,10 +2400,10 @@ public class DynmapCore implements DynmapCommonAPI {
     }
 
     @Override
-    public void processSignChange(int blkid, String world, int x, int y, int z,
+    public void processSignChange(String material, String world, int x, int y, int z,
             String[] lines, String playerid) {
         DynmapPlayer dp = server.getPlayer(playerid);
-        listenerManager.processSignChangeEvent(EventType.SIGN_CHANGE, blkid, world, x, y, z, lines, dp);
+        listenerManager.processSignChangeEvent(EventType.SIGN_CHANGE, material, world, x, y, z, lines, dp);
     }
     
     public MapStorage getDefaultMapStorage() {

--- a/DynmapCore/src/main/java/org/dynmap/common/DynmapListenerManager.java
+++ b/DynmapCore/src/main/java/org/dynmap/common/DynmapListenerManager.java
@@ -30,10 +30,10 @@ public class DynmapListenerManager {
         public void chatEvent(DynmapPlayer p, String msg);
     }
     public interface BlockEventListener extends EventListener {
-        public void blockEvent(int blkid, String w, int x, int y, int z);
+        public void blockEvent(String material, String w, int x, int y, int z);
     }
     public interface SignChangeEventListener extends EventListener {
-        public void signChangeEvent(int blkid, String w, int x, int y, int z, String[] lines, DynmapPlayer p);
+        public void signChangeEvent(String material, String w, int x, int y, int z, String[] lines, DynmapPlayer p);
     }
     public enum EventType {
         WORLD_LOAD,
@@ -105,7 +105,7 @@ public class DynmapListenerManager {
             }
         }
     }
-    public void processBlockEvent(EventType type, int blkid, String world, int x, int y, int z)
+    public void processBlockEvent(EventType type, String material, String world, int x, int y, int z)
     {
         ArrayList<EventListener> lst = listeners.get(type);
         if(lst == null) return;
@@ -114,14 +114,14 @@ public class DynmapListenerManager {
             EventListener el = lst.get(i);
             if(el instanceof BlockEventListener) {
                 try {
-                    ((BlockEventListener)el).blockEvent(blkid, world, x, y, z);
+                    ((BlockEventListener)el).blockEvent(material, world, x, y, z);
                 } catch (Throwable t) {
-                    Log.warning("processBlockEvent(" + type + "," + blkid + "," + world + "," + x + "," + y + "," + z + ") - exception", t);
+                    Log.warning("processBlockEvent(" + type + "," + material + "," + world + "," + x + "," + y + "," + z + ") - exception", t);
                 }
             }
         }
     }
-    public void processSignChangeEvent(EventType type, int blkid, String world, int x, int y, int z, String[] lines, DynmapPlayer p)
+    public void processSignChangeEvent(EventType type, String material, String world, int x, int y, int z, String[] lines, DynmapPlayer p)
     {
         ArrayList<EventListener> lst = listeners.get(type);
         if(lst == null) return;
@@ -130,9 +130,9 @@ public class DynmapListenerManager {
             EventListener el = lst.get(i);
             if(el instanceof SignChangeEventListener) {
                 try {
-                    ((SignChangeEventListener)el).signChangeEvent(blkid, world, x, y, z, lines, p);
+                    ((SignChangeEventListener)el).signChangeEvent(material, world, x, y, z, lines, p);
                 } catch (Throwable t) {
-                    Log.warning("processSignChangeEvent(" + type + "," + blkid + "," + world + "," + x + "," + y + "," + z + ") - exception", t);
+                    Log.warning("processSignChangeEvent(" + type + "," + material + "," + world + "," + x + "," + y + "," + z + ") - exception", t);
                 }
             }
         }

--- a/DynmapCore/src/main/java/org/dynmap/hdmap/TexturePack.java
+++ b/DynmapCore/src/main/java/org/dynmap/hdmap/TexturePack.java
@@ -2603,7 +2603,7 @@ public class TexturePack {
      * @param ps - perspective state
      * @param mapiter - map iterator
      * @param rslt - color result (returned with value)
-     * @param blkid - block ID
+     * @param blk - block state
      * @param lastblocktype - last block ID
      * @param ss - shader state
      */

--- a/DynmapCore/src/main/java/org/dynmap/markers/impl/MarkerSignManager.java
+++ b/DynmapCore/src/main/java/org/dynmap/markers/impl/MarkerSignManager.java
@@ -28,7 +28,7 @@ public class MarkerSignManager {
     
     private static class SignListener implements DynmapListenerManager.SignChangeEventListener, Runnable {
         @Override
-        public void signChangeEvent(int blkid, String wname, int x, int y, int z, String[] lines, DynmapPlayer p) {
+        public void signChangeEvent(String material, String wname, int x, int y, int z, String[] lines, DynmapPlayer p) {
             if(mgr == null)
                 return;			
 			

--- a/DynmapCoreAPI/src/main/java/org/dynmap/DynmapCommonAPI.java
+++ b/DynmapCoreAPI/src/main/java/org/dynmap/DynmapCommonAPI.java
@@ -135,7 +135,7 @@ public interface DynmapCommonAPI {
     public boolean testIfPlayerInfoProtected();
     /**
      * Process sign change
-     * @param blkid - block ID
+     * @param material - block's Material enum value as a string
      * @param world - world name
      * @param x - x coord
      * @param y - y coord
@@ -143,5 +143,5 @@ public interface DynmapCommonAPI {
      * @param lines - sign lines (input and output)
      * @param playerid - player ID
      */
-    public void processSignChange(int blkid, String world, int x, int y, int z, String[] lines, String playerid);
+    public void processSignChange(String material, String world, int x, int y, int z, String[] lines, String playerid);
 }

--- a/spigot/src/main/java/org/dynmap/bukkit/DynmapPlugin.java
+++ b/spigot/src/main/java/org/dynmap/bukkit/DynmapPlugin.java
@@ -100,6 +100,7 @@ import org.dynmap.common.DynmapListenerManager.EventType;
 import org.dynmap.hdmap.HDMap;
 import org.dynmap.markers.MarkerAPI;
 import org.dynmap.modsupport.ModSupportImpl;
+import org.dynmap.renderer.DynmapBlockState;
 import org.dynmap.utils.MapChunkCache;
 import org.dynmap.utils.Polygon;
 import org.dynmap.utils.VisibilityLimit;
@@ -351,7 +352,7 @@ public class DynmapPlugin extends JavaPlugin implements DynmapAPI {
                             Block b = evt.getBlock();
                             if(b == null) return;   /* Work around for stupid mods.... */
                             Location l = b.getLocation();
-                            core.listenerManager.processBlockEvent(EventType.BLOCK_BREAK, b.getType().getId(),
+                            core.listenerManager.processBlockEvent(EventType.BLOCK_BREAK, b.getType().name(),
                                 getWorld(l.getWorld()).getName(), l.getBlockX(), l.getBlockY(), l.getBlockZ());
                         }
                     }, DynmapPlugin.this);
@@ -366,7 +367,7 @@ public class DynmapPlugin extends JavaPlugin implements DynmapAPI {
                             DynmapPlayer dp = null;
                             Player p = evt.getPlayer();
                             if(p != null) dp = new BukkitPlayer(p);
-                            core.listenerManager.processSignChangeEvent(EventType.SIGN_CHANGE, b.getType().getId(),
+                            core.listenerManager.processSignChangeEvent(EventType.SIGN_CHANGE, b.getType().name(),
                                 getWorld(l.getWorld()).getName(), l.getBlockX(), l.getBlockY(), l.getBlockZ(), lines, dp);
                         }
                     }, DynmapPlugin.this);
@@ -1631,9 +1632,9 @@ public class DynmapPlugin extends JavaPlugin implements DynmapAPI {
         }));
     }
     @Override
-    public void processSignChange(int blkid, String world, int x, int y, int z,
+    public void processSignChange(String material, String world, int x, int y, int z,
             String[] lines, String playerid) {
-        core.processSignChange(blkid, world, x, y, z, lines, playerid);
+        core.processSignChange(material, world, x, y, z, lines, playerid);
     }
     
     Polygon getWorldBorder(World w) {

